opentelemetry-sdk/src/opentelemetry/sdk/trace/export/experimental/processor.py

```diff
@@ -0,0 +1,63 @@
+import typing
+
+from opentelemetry.context import Context
+from opentelemetry.sdk.trace import ReadableSpan, SpanProcessor, Span
+from opentelemetry.sdk.trace.export import SpanExporter, SpanExportResult
+from opentelemetry.sdk.trace.export.experimental.timer import TimerABC, PeriodicTimer
+from opentelemetry.sdk.trace.export.experimental.util import SpanAccumulator
+
+
+class BatchSpanProcessor(SpanProcessor):
+    """
+    A SpanProcessor that sends spans in batches on an interval or when a maximum number of spans has been reached,
+    whichever comes first.
+    """
+
+    def __init__(
+        self,
+        exporter: SpanExporter,
+        max_batch_size: int = 1024,
+        interval_sec: int = 4,
+        timer: typing.Optional[TimerABC] = None,
+    ):
+        self._exporter = exporter
+        self._max_batch_size = max_batch_size
+        self._accumulator = SpanAccumulator()
+        self._timer = timer or PeriodicTimer(interval_sec)
+        self._timer.set_callback(self._export)
+        self._timer.start()
+
+    def on_start(self, span: Span, parent_context: typing.Optional[Context] = None) -> None:
+        pass
+
+    def on_end(self, span: ReadableSpan) -> None:
+        """
+        This method must be extremely fast. It adds the span to the accumulator for later sending and pokes the timer
+        if the number of spans waiting to be sent has reached the maximum batch size.
+        """
+        size = self._accumulator.push(span)
```

Comment:
Good question -- this implementation means that if you try to push a lot of spans through simultaneously with multiple threads, exports may contain batches of somewhat more or fewer than the max_batch_size number of spans. I figure this is okay at this stage of development.

But we may end up wanting to do something like: lock, append the span, test for length, generate something like a flush request, then release the lock. This is a fair amount of complexity though, which I'm trying to avoid. I'm hoping we can arrive at a solution that's maximally simple (suggestions always welcome).

Reply:
I have updated the [Accumulator](https://github.com/open-telemetry/opentelemetry-python/pull/3465/files#diff-766b009596e287c573fa451996151d2a984bea543e008c803ff51fe474e22e6bR8) to do batching, so we should have uniform batch sizes now.